DEVELOPMENT.md

```diff
@@ -239,3 +239,55 @@ In our CI/CD pass, we use golint, so feel free to install and run it locally bef
 ```
 go get golang.org/x/lint/golint
 ```
+
+# For People Doing a Release
+
+This section is written primarily for maintainers, so we don't forget how to initiate and complete a release. If you're not a maintainer and you're curious how we release new versions of Athens, read on!
+
+Take a look at our [releases page](https://github.com/gomods/athens/releases). This is how we release official new builds of Athens. Almost all of the release process is automatic.
+
+## Code freeze
+
+The first step to a release is a code freeze. This is a period of time during which we don't merge anything to the `master` branch. The code in `master` is essentially a release candidate (we don't cut a new branch for RC's at the moment) to test. We'll push bugfixes to `master` only if needed.
+
+## Release branch
+
+Right before we un-freeze the `master` branch, we will cut a new branch called `release-v1.X.Y`. Make sure the branch name starts with `release-v`, followed by the [semver](https://semver.org) release number.
+
+### Note about the release number
+
+We follow the [semver](https://semver.org) convention for our release numbers. At the moment, we're not at `v1`, so release versions are `v0.x.y`. `x` is the `MINOR` number, and `y` is the `MAJOR` number. You'll need to decide which one to update.
```

Comment:
This isn't right. `x` is MINOR, `y` is PATCH. MAJOR is `0` in this case (according to semver), we should stick to that.

Reply:
Good catch, thanks @twexler